### DISCUSSION TITLE
Supporting non-standard environments

### DISCUSSION THREAD

**Discussion by willruggiano** [biomejs/biome#916] [General]
Related to https://github.com/biomejs/biome/issues/636

NixOS is my primary interest, since that is the operating system I use on all of my machines.

The problem is basically this: NixOS doesn't work out of the box with vendored binaries. Typically this is not a problem (especially in JS/TS projects). However, since biome is written in Rust and compiled to a native binary (library? not sure), simply `pnpm add -g @biomejs/biome`ing will not work (see error in #636).

I'm honestly not sure what the recommended approach should be here, in the general (not specific to biome) sense. Many, many dev dependencies recommend installing the tool globally (prisma being another example). Personally, I _prefer_ having these dependencies explicitly included in my project's package.json. I feel like I am probably in the majority there. (but doesn't really matter).

Anyways, I'd like to be able to `pnpm add -d` biome to my project. BUT I'd like to say "don't download your own binary/lib, use the one I give you". Lots of possible ways to do this; my preference would be environment variable (which is how prisma lets you do it; `PRISMA_SCHEMA_ENGINE_BINARY`).

It's not the biggest deal in the world, but would make my (and probably others, :cough: @FMGordillo :cough:) life a little easier :)

**Comment by willruggiano** [comment]
Obviously open to your own suggestions/alternatives

**Comment by nhedger** [comment]
Not sure if I'm missing something here, but the sole purpose of the `@biomejs/biome` package is to require one of the platform-specific packages (e.g. `@biomejs/cli-linux-x64`), which only contains the `biome` binary.

What's the use case for adding this package to your dev dependencies if you cannot use the binary anyway ?

**Reply by willruggiano** [reply]
For the rest of the project's contributors

**Comment by willruggiano** [comment]
WAIT. Sorry, I should've looked at the code before asking this question.

https://github.com/biomejs/biome/blob/main/packages/%40biomejs/biome/bin/biome#L22

**Reply by FMGordillo** [reply]
I don't have much experience yet with Nix, and I've faced the same issue that you've mentioned with Prisma but I couldn't get it to run.

If you have a `shell.nix`, script or something similar to learn from it, I'd be very grateful with you!

**Reply by willruggiano** [reply]
Check out this thread; https://github.com/prisma/prisma/issues/3026#issuecomment-1811845865

**Reply by willruggiano** [reply]
for biome it will be similar;
```shell
export BIOME_BINARY="${pkgs.biome}/bin/biome"
```
